### PR TITLE
[mesheryctl] bug: stop restart on non-confirmation

### LINKED ISSUE #5043: [mesheryctl] Negative confirmation to `restart` still goes through with start operation
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior

`mesheryctl system restart` asks confirmation to stop the deployments. Answering `n` to this confirmation cancells the `stop` operation but still goes through with the `start` operation because this confirmation is handled in `stop`.

#### Desired Behavior

The confirmation should be handled by `restart` and passed to `stop` and `restart` must be cancelled on negative confirmation

#### Screenshots/Logs

```
$ ./mesheryctl system restart
Restarting Meshery...
Meshery deployments will be deleted from your cluster. Are you sure you want to continue [y/n]? n
Stop aborted.
Starting Meshery...
```
#### Environment

- Host OS: Linux
- Meshery Server Version: v0.6.0-rc-4
- Meshery Client Version: v0.6.0-rc-4
- Platform: Kubernetes

---

#### Contributor Resources
- [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by pottekkat**
Signed-off-by: Navendu Pottekkat <navendupottekkat@gmail.com>

**Description**

This PR fixes #5043

Updates `system restart` to stop executing if user replies "no" on confirmation.

Current behaviour of `system restart`:

1. When Meshery is not running:

```
mesheryctl system restart
Starting Meshery...

Meshery deployed on Kubernetes.
Meshery is starting...
Opening Meshery (http://localhost:9081) in browser.
```

2. On replying "no" when asking for confirmation (concern of the linked issue):

```
mesheryctl system restart
Meshery deployments will be deleted from your cluster. Are you sure you want to continue [y/n]? n
Restart aborted.
```

3. On replying "yes" when asking for confirmation:

```
mesheryctl system restart
Meshery deployments will be deleted from your cluster. Are you sure you want to continue [y/n]? y
Restarting Meshery...
Stopping Meshery...
Deleting Meshery Namespace...
Meshery is stopped.
Starting Meshery...
```

4. On passing the `--yes` flag:

```
mesheryctl system restart --yes
Restarting Meshery...
Stopping Meshery...
Deleting Meshery Namespace...
Meshery is stopped.
Starting Meshery...
```

- [x] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/internal/cli/root/system/restart.go [hunk 1]
```diff
@@ -87,6 +85,24 @@ func restart() error {
...
+			} else {
+				// ask user for confirmation
+				userResponse = utils.AskForConfirmation("Meshery deployments will be deleted from your cluster. Are you sure you want to continue")
+			}
+			if !userResponse {
+				log.Info("Restart aborted.")
+				return nil
+			}
+			// skips asking for confirmation in the stop() function
+			utils.SilentFlag = true
```
Comment (hexxdump) [inline mesheryctl/internal/cli/root/system/restart.go:101]: I feel, we should let start command ask the confirmation related to endpoint again, we should not assume the response related to k8s cluster is same as response related to endpoint.

Comment (hexxdump) [inline mesheryctl/internal/cli/root/system/restart.go:101]: I feel, we should let start command ask the confirmation related to endpoint again, we should not assume the response related to k8s cluster is same as response related to endpoint.

Comment (hexxdump) [inline mesheryctl/internal/cli/root/system/restart.go:101]: I feel, we should let start command ask the confirmation related to endpoint again, we should not assume the response related to k8s cluster is same as response related to endpoint.

Comment (hexxdump) [inline mesheryctl/internal/cli/root/system/restart.go:101]: @navendu-pottekkat,
I feel, we should let start command ask the confirmation related to endpoint again, we should not assume the response related to k8s cluster is same as response related to endpoint.

Comment (hexxdump) [inline mesheryctl/internal/cli/root/system/restart.go:101]: @navendu-pottekkat,
I feel, we should let start command ask the confirmation related to endpoint again, we should not assume the response related to k8s cluster is same as response related to endpoint.

Comment (navendu-pottekkat) [reply mesheryctl/internal/cli/root/system/restart.go:101]: Got it. Only set the `SlientFlag` if the user passes `--yes`. I will update it to ensure that we don't change the flag value unintentionally for the `start` command.

**Comment by warunicorn19** [comment]
hey @navendu-pottekkat let's discuss this PR on the meshery development call.

**Comment by navendu-pottekkat** [comment]
@hexxdump Updated so that the confirmation in `start` is only skipped when the user passes the flag and the `stop` command does not unintentionally skip it.

**Comment by piyushsingariya** [comment]
We can ignore Integration Tests, it will be soon fixed by @gr455

File: mesheryctl/internal/cli/root/system/restart.go [hunk 2]
```diff
@@ -87,10 +89,32 @@ func restart() error {
...
+				// ask user for confirmation
+				userResponse = utils.AskForConfirmation("Meshery deployments will be deleted from your cluster. Are you sure you want to continue")
+			}
+			if !userResponse {
+				log.Info("Restart aborted.")
+				return nil
+			}
+			// skips asking for confirmation in the stop() function
+			silentFlagSet = utils.SilentFlag
+			utils.SilentFlag = true
```
[the remark this answers is not in the document]
Comment (navendu-pottekkat) [reply mesheryctl/internal/cli/root/system/restart.go]: Updated PR with this change.

**Pull request merged**